src/main/scala/com/gu/fastly/Lambda.scala

```diff
@@ -23,6 +23,60 @@ class Lambda {
   private val cloudWatchClient = AmazonCloudWatchClientBuilder.defaultClient
   private val snsClient = AmazonSNSClientBuilder.defaultClient
 
+  private def raiseAllThePurges(event: Event): Boolean = {
+    val contentType = extractUpdateContentType(event)
+    val purgeType = if (event.eventType == EventType.Delete) {
+      Hard
+    } else {
+      Soft
+    }
+    val aliasPaths = extractAliasPaths(event)
+    val dotcomCanonicalPurge = sendFastlyPurgeRequest(event.payloadId, purgeType, config.fastlyDotcomServiceId, makeDotcomSurrogateKey(event.payloadId), config.fastlyDotcomApiKey, contentType)
+    val jsonCanonicalPurge = sendFastlyPurgeRequestForAjaxFile(event.payloadId, contentType)
+    val mapiCanonicalPurge = sendFastlyPurgeRequest(event.payloadId, purgeType, config.fastlyMapiServiceId, makeMapiSurrogateKey(event.payloadId), config.fastlyMapiApiKey, contentType)
+    val aliasPurges = aliasPaths.fold(Seq.empty[Boolean])({
+      _.map { aliasPath =>
+        val dotcomAliasPurge = sendFastlyPurgeRequest(aliasPath, purgeType, config.fastlyDotcomServiceId, makeDotcomSurrogateKey(aliasPath), config.fastlyDotcomApiKey, contentType)
+        val jsonAliasPurge = sendFastlyPurgeRequestForAjaxFile(aliasPath, contentType)
+        val mapiAliasPurge = sendFastlyPurgeRequest(aliasPath, purgeType, config.fastlyMapiServiceId, makeMapiSurrogateKey(aliasPath), config.fastlyMapiApiKey, contentType)
+        dotcomAliasPurge && jsonAliasPurge && mapiAliasPurge
+      }
+    })
+    // all or nothing result
+    dotcomCanonicalPurge && jsonCanonicalPurge && mapiCanonicalPurge && !aliasPurges.contains(false)
+  }
+
+  private def makeContentDecachedEventsFromCrierEvent(crierEvent: com.gu.crier.model.event.v1.Event): Seq[ContentDecachedEvent] = {
+    // if an update or delete from Crier features a content item with
+    // aliasPaths, we must raise de-cache events for the current path and
+    // all aliases
+    val fastlyEventType = crierEvent.eventType match {
+      case EventType.Delete => com.gu.fastly.model.event.v1.EventType.Delete
+      case _ => com.gu.fastly.model.event.v1.EventType.Update
+    }
+    val contentType = extractUpdateContentType(crierEvent)
+    val aliasPaths = extractAliasPaths(crierEvent)
+    // Always raise an event for the current canonical path (sent via the
+    // crier event's payloadId)
+    val canonicalDecachedEvent = ContentDecachedEvent(
+      crierEvent.payloadId,
+      fastlyEventType,
+      contentType
+    )
+    // If we have aliasPaths, generate their de-cached events too.
+    // Note: If there are no aliasPaths we'll just emit a Seq of one
+    // containing the above canonical de-cached event
+    aliasPaths.fold(Seq(canonicalDecachedEvent)) { paths =>
```

Comment:
Again, could remove duplication by combining the `aliasPaths` and `crierEvent.payloadId`.

Reply:
Yep - think I've done this.